### ISSUE TITLE
Integrate go report card service to give a check and add a badges for the project 

### ISSUE THREAD

**Issue by yanghua** [edgexfoundry/edgex-go#280]
here is a report about edgex-go : https://goreportcard.com/report/github.com/edgexfoundry/edgex-go

cc @jpwhitemn @drasko

**Comment by tsconn23** [comment]
I don't particularly see an action item here. It's a nice tool that we can run on the fly but this looks to me like an FYI rather than an issue to be solved. Could be shared via EdgeX-GoLang mailing list or, alternatively, RocketChat.

**Comment by drasko** [comment]
@yanghua @tsconn23 I would like that we add this badge to our GitHub README.

@yanghua we need too do the linting first and push this to A+.

**Comment by tsconn23** [comment]
I do not want a series of tiny code formatting changes interleaving with PRs to support the Delhi roadmap.

**Comment by drasko** [comment]
@tsconn23 are you proposing that we whould reject this task and close the issue?

Our code is not linted: https://goreportcard.com/report/github.com/edgexfoundry/edgex-go#golint and having a badge on the GitHub page that shows A+ would be more attractive for contributors.

How do you propose that we do these code improvements, which are not big issues/milestones for Delhi? We just reject them and never improve code quality?

**Comment by tsconn23** [comment]
@drasko Taking examples from #194 and #198 PRs have been submitted toward California goals which include refactoring that moves us toward the goals outlined in these recommendations. Why can't this issue be addressed the same way? 

OR, if you like, why not close this umbrella topic (which I maintain would be more relevant as a topic on a mailing list or Rocket Chat) and create an issue for each file called out in the linting step on the report card site and address a specific PR for each file being cleaned up? I do not want 30 PRs tied to this one issue and I do not want one massive PR addressing all of the report card edits.

**Comment by yanghua** [comment]
I think open a issue is also a way of discussing. 

Most popular Go projects have this badge, such as : 

* [openfaas](https://github.com/openfaas/faas)
* [influxdb](https://github.com/influxdata/influxdb)
* [beego](https://github.com/astaxie/beego)
* ....

Does this a disincentive problem need to be discussed? Actually, I think it is an obvious improvement for EdgeX.
As @drasko said, this is good for EdgeX's code quality. If a stranger did not know about EdgeX. Some badges would be the primary impression, those badges would give the potential contributors confidence.

The first step, I just want to integrate with the go report card and add a badge for the project, not related to refactor the code, it's future work.

However, I think the refactor work should be done to let the badge shows "A+". About linting, we can consult at one TSC meeting to get a suitable time point? 

I think code quality is a continuing process. Firstly, we should make our code looks good for us and strangers(through badge). Then, we can add some build plugins or CI tools to make imperative check for all PRs.

Firstly, I just want to add a badge for EdgeX, then we can discuss how to refactor code.  

cc @tsconn23

**Comment by tsconn23** [comment]
@yanghua Well which one is it -- Add the badges or address the linting? The subject line of this issue covers both, but there's a new PR #281 that only adds a badge to the Readme. This illustrates what I'm saying. I request to close this issue and create one for the badge. Associate PR #281 to it. Then create an issue for each file in which you plan to address the linting defects.

**Comment by yanghua** [comment]
hi @tsconn23 maybe my expression is poor. The issue's original goal is add a badge. the keyword "integrate" means use go report card's service to scan and give us a report. But the discuss between you and drasko mention the 'linting' so I was brought into it.

if the PR #281 be merged, I will close this issue. And then open another issue to discuss the linting , check tools and code quality. That means this issue just refer to add a badge about go report card.

Do you agree with me? Or I change the issue's title?

**Comment by tsconn23** [comment]
@yanghua @drasko Once PR #281 is merged, then please close this issue. Going forward, subsequent edits to address the linting should have their own issues. Thanks.

**Comment by yanghua** [comment]
@drasko and @tsconn23 thanks, more discussion about code quality and code linting we can go into issue #286 , closing this issue mapping the merged PR #281
